Source/WTF/wtf/glib/RunLoopGLib.cpp

```diff
@@ -141,18 +146,43 @@ RunLoop::CycleResult RunLoop::cycle(RunLoopMode)
     return CycleResult::Continue;
 }
 
-void RunLoop::observe(const RunLoop::Observer& observer)
+void RunLoop::observeEvent(const RunLoop::EventObserver& observer)
 {
-    ASSERT(!m_observers.contains(observer));
-    m_observers.add(observer);
+    ASSERT(!m_eventObservers.contains(observer));
+    m_eventObservers.add(observer);
+}
+
+void RunLoop::observeActivity(const RunLoop::ActivityObserver& observer)
+{
+    ASSERT(!m_activityObservers.contains(observer));
+    m_activityObservers.append(observer);
+
+    std::ranges::sort(m_activityObservers, std::ranges::less { }, &RunLoop::ActivityObserver::order);
+}
+
+void RunLoop::unobserveActivity(const RunLoop::ActivityObserver& observer)
+{
+    ASSERT(m_activityObservers.contains(observer));
+    m_activityObservers.removeFirst(observer);
+}
+
+void RunLoop::notifyActivity(Activity activity)
+{
+    if (m_activityObservers.isEmpty())
+        return;
+
+    for (const auto& observer : m_activityObservers) {
```

Comment:
for loop will do nothing if vector is empty so we don't need the early return, I think.